uaa_client/authentication.py

```diff
@@ -79,52 +79,69 @@ class UaaBackend(ModelBackend):
     so that the superclass can provide all authorization methods.
     '''
 
-    @staticmethod
-    def get_user_by_email(email):
+    @classmethod
+    def get_user_by_email(cls, email):
         '''
         Return a :class:`django.contrib.auth.models.User` with the given
         email address. If no user can be found, return ``None``.
 
         The default implementation attempts to find an existing user with
         the given case-insensitive email address. If no such user exists,
-        ``None`` is returned.
+        :meth:`should_create_user_for_email` is consulted to determine
+        whether the user should be auto-created; if so,
+        :meth:`create_user_with_email` is used to auto-create the user.
+        Otherwise, ``None`` is returned.
 
         Subclasses may override this method to account for different kinds
         of security policies for logins.
         '''
+
         try:
             return User.objects.get(email__iexact=email)
         except User.DoesNotExist:
-            return UaaBackend.consider_a_new_user(email)
+            if cls.should_create_user_for_email(email):
+                return cls.create_user_with_email(email)
+            logger.info(
+                'User with email {} does not exist and is not '
+                'approved for auto-creation'.format(email)
+            )
+            return None
         except MultipleObjectsReturned:
             logger.warning(
                 'Multiple users with email {} exist'.format(email)
             )
             return None
 
-    @staticmethod
-    def consider_a_new_user(email):
+    @classmethod
+    def should_create_user_for_email(cls, email):
```

Comment:
I thought `consider_a_new_user()` was a kind of vague name but I couldn't think of a better one (`maybe_create_a_new_user` sounded too indecisive), so instead I separated the functionality out into two methods with more concise names.